pkg/controller/lambda/function/setup.go

```diff
@@ -0,0 +1,499 @@
+
+	"github.com/crossplane/provider-aws/apis/lambda/v1alpha1"
+	svcapitypes "github.com/crossplane/provider-aws/apis/lambda/v1alpha1"
+	aws "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+// SetupFunction adds a controller that reconciles Function.
+func SetupFunction(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(v1alpha1.FunctionGroupKind)
+	opts := []option{
+		func(e *external) {
+			e.preObserve = preObserve
+			e.postObserve = postObserve
+			e.postCreate = postCreate
+			e.preDelete = preDelete
+			e.isUpToDate = isUpToDate
+			u := &updater{client: e.client}
+			e.update = u.update
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&v1alpha1.Function{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.FunctionGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.Function, obj *svcsdk.GetFunctionInput) error {
+	obj.FunctionName = aws.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.Function, resp *svcsdk.GetFunctionOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+	switch aws.StringValue(resp.Configuration.State) {
+	case string(svcapitypes.State_Active):
+		cr.SetConditions(xpv1.Available())
+	case string(svcapitypes.State_Pending):
+		cr.SetConditions(xpv1.Creating())
+	case string(svcapitypes.State_Failed):
+		cr.SetConditions(xpv1.Unavailable())
+	case string(svcapitypes.State_Inactive):
+		cr.SetConditions(xpv1.Unavailable())
+	}
+	return obs, nil
+}
+
+func postCreate(_ context.Context, cr *svcapitypes.Function, resp *svcsdk.FunctionConfiguration, cre managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+	meta.SetExternalName(cr, aws.StringValue(resp.FunctionName))
+	cre.ExternalNameAssigned = true
+	return cre, nil
+}
+
+func preDelete(_ context.Context, cr *svcapitypes.Function, obj *svcsdk.DeleteFunctionInput) (bool, error) {
+	obj.FunctionName = aws.String(meta.GetExternalName(cr))
+	return false, nil
+}
+
+// nolint:gocyclo
+func isUpToDate(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) (bool, error) {
+
+	// Compare CODE
+	if !isUpToDateCode(cr, obj) {
+		return false, nil
+	}
+
+	// Compare CONFIGURATION
+	if strings.Compare(aws.StringValue(cr.Spec.ForProvider.Description), aws.StringValue(obj.Configuration.Description)) != 0 {
```

Comment:
```suggestion
	if aws.StringValue(cr.Spec.ForProvider.Description) != aws.StringValue(obj.Configuration.Description) {
```

Does `strings.Compare` have an advantage over equality check in this context? Seems like it does an equality check and then another check for lexicographically order, which is not necessary in this context.